ui/packages/catalog-ui-search/src/main/webapp/react-component/location/base.line.js

```diff
@@ -150,6 +180,36 @@ class BaseLine extends React.Component {
       return false
     }
   }
+  convertWkt(value, numCoords) {
+    const coordinatePairs = value.match(coordinatePairRegex)
+    if (!coordinatePairs || coordinatePairs.length < numCoords) {
+      return value
+    }
+    const coordinates = coordinatePairs.map(coord => coord.replace(' ', ','))
+    return `[[${coordinates.join('],[')}]]`
+  }
+  convertMultiWkt(isPolygon, value) {
+    if (isPolygon && !value.endsWith(')))')) {
+      return value
+    } else if (!value.endsWith('))')) {
+      return value
+    }
+    const splitter = isPolygon ? '))' : ')'
+    const numPoints = isPolygon ? 4 : 2
+    let shapes = value
+      .split(splitter)
+      .map(shape => shape.match(coordinatePairRegex))
+    shapes = shapes
+      .filter(shape => shape !== null && shape.length >= numPoints)
+      .map(shape =>
+        shape.map(coordinatePair => coordinatePair.replace(' ', ','))
+      )
+    return shapes.length === 0
+      ? value
+      : shapes.length === 1
+        ? `[[${shapes[0].join('],[')}]]`
+        : `[${shapes.map(shapeCoords => `[[${shapeCoords.join('],[')}]]`)}]`
```

Comment:
This has a lot going on (double \` for instance) and could be pulled into another method